src/middlewared/middlewared/plugins/pool.py

```diff
@@ -301,6 +302,7 @@ class Config:
                 'total_secs_left': None,
             }
         ),
+        Bool('is_upgraded'),
```

Comment:
Does this default to `True`? If so, does this mean that anytime `pool.query` is called, we `subprocess` for every pool on the system?

Reply:
This is `ENTRY` declaration, a schema for a returned pool object, here I declare that `is_upgraded` is boolean and is not required (might be either present or not). We only call subprocess `if context['extra'].get('is_upgraded')` and `extra` is default to an empty dict.